security/pkg/pki/ca/ca.go

```diff
@@ -119,9 +119,18 @@ type IstioCAOptions struct {
 // NewSelfSignedIstioCAOptions returns a new IstioCAOptions instance using self-signed certificate.
 func NewSelfSignedIstioCAOptions(ctx context.Context,
 	rootCertGracePeriodPercentile int, caCertTTL, rootCertCheckInverval, defaultCertTTL,
-	maxCertTTL time.Duration, org string, dualUse bool, namespace string, client corev1.CoreV1Interface,
+	maxCertTTL time.Duration, org string, useCacertsSecretName, dualUse bool, namespace string, client corev1.CoreV1Interface,
 	rootCertFile string, enableJitter bool, caRSAKeySize int,
 ) (caOpts *IstioCAOptions, err error) {
+	// TODO(jaellio): Implement istio-ca-secret to cacerts Secret migration. If only a istio-ca-secret
+	// exists, create a new cacerts secret and load the cacerts data. Currently, if
+	// USE_CACERTS_FOR_SELF_SIGNED_CA is enabled we ignore an existing istio-ca-secret and create a
```

Comment:
We need to create both two secrets named `istio-ca-secret` and `cacerts` to support forward compatibility.  Let's think about whether it is a must

Reply:
Not sure I understand. Are you saying creating both secrets is a must? What do we want this feature flag to mean/ protect?